Lib/STL.hpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * This file is part of the source code of the software program
+ * Vampire. It is protected by applicable
+ * copyright laws.
+ *
+ * This source code is distributed under the licence found here
+ * https://vprover.github.io/license.html
+ * and in the source directory
+ */
+
+#ifndef STL_HPP
+#define STL_HPP
+
+#include <algorithm>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+namespace Lib {
+
+template<class T>
+T* move_to_heap(T t) { return new T(std::move(t)); }
+                
+template< typename T, 
+          std::enable_if_t<!std::is_pointer<T>::value, bool> = true  
+        >
+std::shared_ptr<T> make_shared(T t)
+{ return std::shared_ptr<T>(move_to_heap(std::move(t))); }
```

Comment:
How does this differ from `std::make_shared`?

Reply:
Only in that we don't have to specify the type argument `T` as it's always equal to the function argument. That's quite helpful to not having to repeat long iterator types and similar things.

Reply:
Ahah - it's some special case of the `Args...` constructor, got it.